backend/src/plugins/Automod/triggers/matchMimeType.ts

```diff
@@ -0,0 +1,79 @@
+import { automodTrigger } from "../helpers";
+import * as t from "io-ts";
+import { asSingleLine, messageSummary, verboseChannelMention } from "src/utils";
+import { Snowflake, TextChannel, Util } from "discord.js";
+
+interface MatchResultType {
+  matchedType: string;
+  mode: "blacklist" | "whitelist";
+}
+
+export const MatchMimeTypeTrigger = automodTrigger<MatchResultType>()({
+  configType: t.type({
+    mime_type_blacklist: t.array(t.string),
+    blacklist_enabled: t.boolean,
+    mime_type_whitelist: t.array(t.string),
+    whitelist_enabled: t.boolean,
+  }),
+
+  defaultConfig: {
+    mime_type_blacklist: [],
+    blacklist_enabled: false,
+    mime_type_whitelist: [],
+    whitelist_enabled: false,
+  },
+
+  async match({ context, triggerConfig: trigger }) {
+    if (!context.message) return;
+
+    const { attachments } = context.message.data;
+    if (!attachments) return null;
+
+    for (const attachment of attachments) {
+      const { contentType } = attachment;
+
+      const blacklist = trigger.blacklist_enabled
+        ? (trigger.mime_type_blacklist || []).map(_t => _t.toLowerCase())
+        : null;
+
+      if (contentType && blacklist?.includes(contentType)) {
+        return {
+          extra: {
+            matchedType: contentType,
+            mode: "blacklist",
+          },
+        };
+      }
+
+      const whitelist = trigger.whitelist_enabled
+        ? (trigger.mime_type_whitelist || []).map(_t => _t.toLowerCase())
+        : null;
+
+      if (whitelist && (!contentType || !whitelist.includes(contentType))) {
+        return {
+          extra: {
+            matchedType: contentType || "unknown",
+            mode: "whitelist",
+          },
+        };
+      }
+
+      return null;
+    }
+  },
+
+  renderMatchInformation({ pluginData, contexts, matchResult }) {
+    const { message } = contexts[0];
+    const channel = pluginData.guild.channels.cache.get(message!.channel_id as Snowflake) as TextChannel;
```

Comment:
Besides the unnecessary Snowflake assertion, the channel assertion might not be accurate. E.g., this could have been matched on a ThreadChannel or NewsChannel

```suggestion
    const channel = pluginData.guild.channels.resolve(message!.channel_id) as TextChannel;
```

Reply:
The `as Snowflake` must've been some type assertion I had to do before the full migration was complete.

Reply:
Yeah I'm aware, but it isn't required anymore